sdk/identity/identity/src/credentials/environmentCredential.ts

```diff
@@ -52,17 +57,17 @@ export class EnvironmentCredential implements TokenCredential {
    */
   constructor(options?: TokenCredentialOptions) {
     // Keep track of any missing environment variables for error details
-    this._environmentVarsMissing = AllSupportedEnvironmentVariables.filter(
-      (v) => process.env[v] === undefined
-    );
+
+    const assigned = processEnvVars(AllSupportedEnvironmentVariables).assigned.join(", ");
+    logger.info(`Found the following environment variables: ${assigned}`);
 
     const tenantId = process.env.AZURE_TENANT_ID,
       clientId = process.env.AZURE_CLIENT_ID,
       clientSecret = process.env.AZURE_CLIENT_SECRET;
 
     if (tenantId && clientId && clientSecret) {
       logger.info(
-        `EnvironmentCredential: loaded with tenant ID: ${tenantId}, clientId: ${clientId}`
+        `Invoking ClientSecretCredential with tenant ID: ${tenantId}, clientId: ${clientId} and clientSecret: ${clientSecret}`
```

Comment:
I dont think you want to be printing the client secret. We agreed not to include any secret information in logs.

Reply:
If you want to do this, then there should be a warning or something really clear for users to know that their secrets could be printed to a log file. This mainly poses a security risk if we log this by default.

Reply:
OMG how did I miss this? thank you!

Reply:
I'm thinking on changing this to: `Invoking ClientSecretCredential with tenant ID: ${tenantId}, clientId: ${clientId} and a redacted (but not-empty) clientSecret.` How does that look?

Reply:
Yes I think that message is fine.

Reply:
I decided to go with `[REDACTED]`, as in `Invoking ClientSecretCredential with tenant ID: ${tenantId}, clientId: ${clientId} and clientSecret: [REDACTED]`, so that I could use it more comfortably in other places.